src/dlgConnectionProfiles.cpp

```diff
@@ -628,12 +634,19 @@ void dlgConnectionProfiles::slot_deleteProfile()
 
     QString profile = profiles_tree_widget->currentItem()->data(csmNameRole).toString();
 
+    QDir profileDirContents(mudlet::getMudletPath(mudlet::profileXmlFilesPath, profile));
+    if (profileDirContents.count() == 0) {
```

Comment:
I think you will do better with:
```suggestion
    if (profileDirContents.isEmpty()) {
```
The documentation for this method does note:
> `bool QDir::isEmpty(QDir::Filters filters = Filters(AllEntries | NoDotAndDotDot)) const`
>
> Returns whether the directory is empty.
>
> Equivalent to `count() == 0` with filters `QDir::AllEntries | QDir::NoDotAndDotDot`, but faster as it just checks whether the directory contains at least one entry.
>
> Note: Unless you set the filters flags to include `QDir::NoDotAndDotDot` (as the default value does), no directory is empty.
>
> This function was introduced in Qt 5.9.
>
> See also `count()`, `entryList()`, and `setFilter()`.

Fair enough you are *really* looking for a non-existent directory so perhaps you should go for:
```suggestion
    if (!profileDirContents.exists() || profileDirContents.isEmpty()) {
```